network/src/main/scala/com/linkedin/norbert/network/netty/NettyClusterIoClientComponent.scala

```diff
@@ -35,15 +35,21 @@ trait NettyClusterIoClientComponent extends ClusterIoClientComponent {
 
     def sendMessage[RequestMsg, ResponseMsg](node: Node, request: Request[RequestMsg, ResponseMsg]) {
       if (node == null || request == null) throw new NullPointerException
-
-      val pool = getChannelPool(node)
-      try {
-        pool.sendRequest(request)
-      } catch {
-        case ex: ChannelPoolClosedException =>
-          // ChannelPool was closed, try again
-          sendMessage(node, request)
+      val maxRetries = 3
+      for (_ <- 1 to maxRetries) {
```

Comment:
Please make it as member variable not local function variable. It would be nice we can make it as configurable with default value.